snippets/native-platforms/ionic2/use.md

```diff
@@ -1,98 +1,142 @@
+  callbackURL: Auth0Vars.AUTH0_CALLBACK_URL,
+  packageIdentifier: Auth0Vars.AUTH0_APPLICATION_PACKAGE_NAME
+};
 
-// Avoid name not found warnings
-declare var Auth0: any;
-declare var Auth0Lock: any;
 
 @Injectable()
 export class AuthService {
-
-  jwtHelper: JwtHelper = new JwtHelper();
-  auth0 = new Auth0({clientID: Auth0Vars.AUTH0_CLIENT_ID, domain: Auth0Vars.AUTH0_DOMAIN });
-  lock = new Auth0Lock(Auth0Vars.AUTH0_CLIENT_ID, Auth0Vars.AUTH0_DOMAIN, {
-    auth: {
-      redirect: false,
-      params: {
-        scope: 'openid profile offline_access',
-        device: 'my-device'
-      },
-      sso: false
-    }
-  });
-  storage: Storage = new Storage();
-  refreshSubscription: any;
-  user: Object;
-  zoneImpl: NgZone;
+  jwtHelper = new JwtHelper();
+  auth0 = new Auth0.WebAuth(auth0Config);
+  refreshSubscription$: Subscription;
   accessToken: string;
   idToken: string;
-  
-  constructor(private authHttp: AuthHttp, zone: NgZone) {
-    this.zoneImpl = zone;
-    // Check if there is a profile saved in local storage
-    this.storage.get('profile').then(profile => {
-      this.user = JSON.parse(profile);
-    }).catch(error => {
-      console.log(error);
-    });
+  user: any;
 
-    this.storage.get('id_token').then(token => {
-      this.idToken = token;
-    });
+  constructor(public zone: NgZone) {
+    this.user = this.getStorageVariable('profile');
+    this.idToken = this.getStorageVariable('id_token');
+  }
 
-    this.lock.on('authenticated', authResult => {
-      if (authResult && authResult.accessToken && authResult.idToken) {
-        this.storage.set('access_token', authResult.accessToken);
-        this.storage.set('id_token', authResult.idToken);
-        this.storage.set('refresh_token', authResult.refreshToken);
-        this.accessToken = authResult.accessToken;
-        this.idToken = authResult.idToken;
-
-        // Fetch profile information
-        this.lock.getUserInfo(this.accessToken, (error, profile) => {
-          if (error) {
-            // Handle error
-            alert(error);
-            return;
-          }
-
-          profile.user_metadata = profile.user_metadata || {};
-          this.storage.set('profile', JSON.stringify(profile));
-          this.user = profile;
-        });
+  private getStorageVariable(name) {
+    return JSON.parse(window.localStorage.getItem(name));
+  }
 
-        this.lock.hide();
+  private setStorageVariable(name, data) {
+    window.localStorage.setItem(name, JSON.stringify(data));
+  }
 
-        this.zoneImpl.run(() => this.user = authResult.profile);
-        // // Schedule a token refresh
-        this.scheduleRefresh();
-      }
+  private setIdToken(token) {
+    this.idToken = token;
+    this.setStorageVariable('id_token', token);
+  }
 
-    });    
+  private setAccessToken(token) {
+    this.accessToken = token;
+    this.setStorageVariable('access_token', token);
   }
 
-  public authenticated() { 
+  public isAuthenticated() {
```

Comment:
Checking whether the user is authenticated should now be done by calculating the time at which the access token expires. But not by reading the access token. Rather we should use the `expires_in` value that comes back after authentication. See [here](https://github.com/auth0-samples/auth0-angular-samples/blob/master/00-Centralized-Login/src/app/auth/auth.service.ts#L41) for an example of settings it and [here](https://github.com/auth0-samples/auth0-angular-samples/blob/master/00-Centralized-Login/src/app/auth/auth.service.ts#L59-L60) for an example of using it in `isAuthenticated`.